packages/vite/src/esbuild-resolver.ts

```diff
@@ -30,6 +30,9 @@ export function esBuildResolver(root = process.cwd()): EsBuildPlugin {
         let resolution = await resolverLoader.resolver.resolve(request, defaultResolve(build, kind));
         switch (resolution.type) {
           case 'found':
+            if (resolution.result?.path?.includes('rewritten-app')) {
```

Comment:
@mansona I i think i found the real reason we have the optimizer issue.
We call context.resolve in the default resolve here https://github.com/embroider-build/embroider/pull/1718/commits/0aafe02b99bb2f5a54621c42d6722d789f46871a#diff-a6c8071691a61acc68be864b34d3f338fd92e9bf5217c2d12403ac6029bca885R130 
with the full absolute path.
which then goes to the vite dep:scan plugin which marks it as external here: https://github.com/vitejs/vite/blob/main/packages/vite/src/node/optimizer/scan.ts#L681
so either we do not call that and use e.g resolve from npm or unmark them as external here

Reply:
i also created a PR on vite as i think its a vite bug: https://github.com/vitejs/vite/pull/15402
lets see

Reply:
Btw, this was working in windows without issues... Will verify later again why this is. I suspect the generated path has somr difference in slashes